src/api/cvc4cpp.h

```diff
@@ -523,6 +523,122 @@ struct CVC4_PUBLIC SortHashFunction
   size_t operator()(const Sort& s) const;
 };
 
+/* -------------------------------------------------------------------------- */
+/* Op                                                                     */
+/* -------------------------------------------------------------------------- */
+
+/**
+ * A CVC4 operator term.
+ * An operator term is a term that represents certain operators, instantiated
+ * with its required parameters, e.g., a term of kind BITVECTOR_EXTRACT.
+ */
+class CVC4_PUBLIC Op
+{
+  friend class Solver;
+  friend struct OpHashFunction;
+
+ public:
+  /**
+   * Constructor.
+   */
+  Op();
+
+  // !!! This constructor is only temporarily public until the parser is fully
+  // migrated to the new API. !!!
+  /**
+   * Constructor for a single kind (non-indexed operator).
+   * @param k the kind of this Op
+   */
+  Op(const Kind k);
+
+  // !!! This constructor is only temporarily public until the parser is fully
+  // migrated to the new API. !!!
+  /**
+   * Constructor.
+   * @param k the kind of this Op
+   * @param e the internal expression that is to be wrapped by this term
+   * @return the Term
+   */
+  Op(const Kind k, const CVC4::Expr& e);
+
+  /**
+   * Destructor.
+   */
+  ~Op();
+
+  /**
+   * Syntactic equality operator.
+   * Return true if both operator terms are syntactically identical.
+   * Both operator terms must belong to the same solver object.
+   * @param t the operator term to compare to for equality
+   * @return true if the operator terms are equal
+   */
+  bool operator==(const Op& t) const;
+
+  /**
+   * Syntactic disequality operator.
+   * Return true if both operator terms differ syntactically.
+   * Both terms must belong to the same solver object.
+   * @param t the operator term to compare to for disequality
+   * @return true if operator terms are disequal
+   */
+  bool operator!=(const Op& t) const;
+
+  /**
+   * @return the kind of this operator term
+   */
+  Kind getKind() const;
+
+  /**
+   * @return the sort of this operator term
+   */
+  Sort getSort() const;
+
+  /**
+   * @return true if this operator term is a null term
+   */
+  bool isNull() const;
+
+  /**
+   * Get the indices used to create this Op.
+   * Supports the following template arguments:
+   *   - string
+   *   - Kind
+   *   - uint32_t
+   *   - pair<uint32_t, uint32_t>
+   * Check the Op Kind with getKind() to determine which argument to use.
+   * @return the indices used to create this Op
+   */
+  template <typename T>
+  T getIndices() const;
+
+  /**
+   * @return a string representation of this operator term
+   */
+  std::string toString() const;
+
+  // !!! This is only temporarily available until the parser is fully migrated
+  // to the new API. !!!
```

Comment:
What happens if the Op doesn't have an Expr?

Reply:
Also fixed with a332a47a5880592beb687dae8c90aa870eed316c. If it's only holding a Kind, it prints the Kind's toString. Otherwise it uses the Expr. Those should be the only two valid cases, and otherwise it throws an exception (e.g. if it's a null Op). Does this seem right to you?

In particular, I'm not sure what it should return if it only holds a Kind. Should it print something like `Op(PLUS)` or something like that so it doesn't look exactly the same as a Kind?

Reply:
Sorry, my comment was referencing the wrong line (even though it did also apply to `toString()`, which I would have missed otherwise ;-)

What I actually wanted to ask is what happens in the call to `getExpr()` if the Expr is null -- then we simply return the null Expr, right? This might be a problem.

Reply:
> Does this seem right to you?
yes

Reply:
Ahh that's right, we would just return a null Expr. Maybe an assertion failure would be good there as well? What do you think?

Reply:
Assertion failure sounds good, this is a temporary available function for the parser migration and will be internal in the future.